features/step_definitions/speech_steps.rb

```diff
@@ -61,15 +61,15 @@
 
 When /^I draft a new speech "([^"]*)" relating it to the policies "([^"]*)" and "([^"]*)"$/ do |title, first_policy, second_policy|
   begin_drafting_speech title: title
+  # @policies is populated by PolicyTaggingHelpers#stub_publishing_api_policies
   select first_policy, from: "Policies"
   select second_policy, from: "Policies"
   click_button "Save"
 end
 
 Then /^the published speech should remain unchanged$/ do
-  visit public_document_path(@speech)
-  assert page.has_css?('h1', text: @speech.title)
-  assert page.has_css?('.body', text: @speech.body)
+  latest_edition = @speech.latest_edition
```

Comment:
perhaps just a small rename here to be clear that it is a `draft_edition`. Or even cutting it out completely? We have a unit test on Edition that sort of does something similar: https://github.com/alphagov/whitehall/blob/master/test/unit/edition_test.rb#L43
Maybe that's enough of a test for this kind of scenario?